extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java

```diff
@@ -267,19 +267,22 @@ void builtinCallbackArguments(BuildProducer<CallbackArgumentBuildItem> providers
     void collectGlobalErrorHandlers(BeanArchiveIndexBuildItem beanArchiveIndex,
             BeanDiscoveryFinishedBuildItem beanDiscoveryFinished,
             BuildProducer<GlobalErrorHandlersBuildItem> globalErrorHandlers,
+            BuildProducer<UnremovableBeanBuildItem> unremovableBean,
             CallbackArgumentsBuildItem callbackArguments,
             TransformedAnnotationsBuildItem transformedAnnotations) {
 
         IndexView index = beanArchiveIndex.getIndex();
 
         // Collect global error handlers, i.e. handlers that are not declared on an endpoint
         Map<DotName, GlobalErrorHandler> globalErrors = new HashMap<>();
+        Set<DotName> unremovableBeanClasses = new HashSet<>();
         for (BeanInfo bean : beanDiscoveryFinished.beanStream().classBeans()) {
             ClassInfo beanClass = bean.getTarget().get().asClass();
             if (beanClass.declaredAnnotation(WebSocketDotNames.WEB_SOCKET) == null
                     && beanClass.declaredAnnotation(WebSocketDotNames.WEB_SOCKET_CLIENT) == null) {
-                for (Callback callback : findErrorHandlers(Target.UNDEFINED, index, bean, beanClass, callbackArguments,
-                        transformedAnnotations, null)) {
+                List<Callback> errorHandlers = findErrorHandlers(Target.UNDEFINED, index, bean, beanClass,
```

Comment:
This is a good catch! Previously, we only marked those classes that declare a global error handler but have no scope as unremovable. I think that we should remove this line as well: https://github.com/quarkusio/quarkus/blob/3.25.0/extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java#L192